From 3d0f94eeb871fd12d69fb6e274a06d0fc375d5fa Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 18 Mar 2025 09:30:52 +0100
Subject: [PATCH 813/841] feat(stm32mp2-fdts): add fixed IO compensation on
 vddio regulators for stm32mp211

IO compensation could alter duty cycle of high frequency output signal
on stm32mp25 SoC. To avoid such issue, fixed IO compensation is used.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 22d0762c548e7813b22427cf31d1d4f2b77c73b8 feat(stm32mp2-fdts): add fixed IO compensation on vddio regulators for stm32mp211]
Change-Id: I1bd056bb10f1d4f76ec7324a86018f48fd0a9568
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/444758
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 fdts/stm32mp211.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/fdts/stm32mp211.dtsi b/fdts/stm32mp211.dtsi
index 4cf300fe4..700ced55f 100644
--- a/fdts/stm32mp211.dtsi
+++ b/fdts/stm32mp211.dtsi
@@ -337,18 +337,22 @@
 
 			vddio1: vddio1 {
 				regulator-name = "vddio1";
+				st,iocomp = <8 7>;
 			};
 
 			vddio2: vddio2 {
 				regulator-name = "vddio2";
+				st,iocomp = <8 7>;
 			};
 
 			vddio3: vddio3 {
 				regulator-name = "vddio3";
+				st,iocomp = <8 7>;
 			};
 
 			vddio: vddio {
 				regulator-name = "vddio";
+				st,iocomp = <8 7>;
 			};
 		};
 
-- 
2.47.2

